### PR TITLE
Add "How this works" section to the README

### PR DESCRIPTION
**Pull request by SeanKilleen**
I was interested in how up-for-grabs is built and generates its data for Github pages -- I think there are some really helpful patterns here.

After @dahlbyk was kind enough to respond to my tweets, I figured I'd capture it in the README so that others could get some quick insight into the techniques used.

I think I've got it (mostly) correct; any questions / corrections / edits are welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,3 +70,13 @@ docker-compose up
 ```
 
 Then open your browser to `localhost:4000` to view the site.
+
+## How This All Works
+We use a few great features of Jekyll and Github Pages to host this entire site for free.
```
Comment (shiftkey) [inline README.md]: s/Github/GitHub/

File: README.md [hunk 2]
```diff
@@ -70,3 +70,13 @@ docker-compose up
...
+
+* We are using a file called `scripts.html` which is an include file. When we publish the site, Jekyll uses this to concatenate generate the scripts that will actually be downloaded when a user visits the site.
+* Within the `scripts.html` template, we're referencing `site.data.projects`, which gives us access to the entire directory of `.yml` files as a set of variables. We combine this with the `jsonify` liquid transformer to turn them into a JSON object. This means that the raw data is generated once, so it downloads very quickly as static data.
+* When the web page is opened, our startup JS processes each project in the array of projects and pushes it into the array that the rest of the site uses.
+* We use [travis-ci](https://travis-ci.org/up-for-grabs/up-for-grabs.net) to run a custom ruby script, `cibuild`, that checks all the `.yml` files to make sure they can be appropriately parsed. This makes sure we don't merge any incorrectly formed project files.
+
+What this means is that, when a pull request is merged, Github Pages automatically builds the site via Jekyll and publishes it to our Github -- no database or hosting needed. (Thanks, Github!)
```
Comment (shiftkey) [inline README.md]: s/Github/GitHub/

**Review by shiftkey — CHANGES_REQUESTED** [review]
Just some typos - everything else is :gem:

**Comment by SeanKilleen** [comment]
@shiftkey fixed Github -> GitHub. Should be good to go now. :) Thanks!

**Review by shiftkey — APPROVED** [review]

**Comment by shiftkey** [comment]
@SeanKilleen thanks! 💖

**Pull request merged**